src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -544,6 +566,48 @@ zocl_load_sect(struct drm_zocl_dev *zdev, struct axlf *axlf,
 	case BITSTREAM_PARTIAL_PDI:
 		ret = zocl_load_partial(zdev, section_buffer, size);
 		break;
+	case PARTITION_METADATA:
+		if(zdev->partial_overlay_id != -1 && axlf->m_header.m_mode == XCLBIN_PR) {
+			err = of_overlay_remove(&zdev->partial_overlay_id);
+			if (err < 0) {
+				DRM_WARN("Failed to delete rm overlay (err=%d)\n", err);
+				return err;
+			}
+			zdev->partial_overlay_id = -1;
+		} else if(zdev->full_overlay_id != -1 && axlf->m_header.m_mode == XCLBIN_FLAT){
+			err = of_overlay_remove_all();
+			if (err < 0) {
+				DRM_WARN("Failed to delete static overlay (err=%d)\n", err);
+				return err;
+			}
+			zdev->partial_overlay_id = -1;
+			zdev->full_overlay_id = -1;
+		}
+
+		bsize = zocl_read_sect(BITSTREAM, &bsection_buffer, axlf, xclbin);
+		bo = zocl_drm_create_bo(zdev->ddev, bsize, ZOCL_BO_FLAGS_CMA);
+		if (IS_ERR(bo))
+			return -1;
```

Comment:
do we need to free section buffer before bail out?

Reply:
fixed